src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -353,20 +353,14 @@ else if (exerciseWithOnlyTitle instanceof ProgrammingExercise) {
         Submission originalSubmission = complaint.getResult().getSubmission();
         if (originalSubmission != null) {
             Submission submissionWithOnlyId;
-            if (originalSubmission instanceof TextSubmission) {
-                submissionWithOnlyId = new TextSubmission();
-            }
-            else if (originalSubmission instanceof ModelingSubmission) {
-                submissionWithOnlyId = new ModelingSubmission();
-            }
-            else if (originalSubmission instanceof FileUploadSubmission) {
-                submissionWithOnlyId = new FileUploadSubmission();
-            }
-            else if (originalSubmission instanceof ProgrammingSubmission) {
-                submissionWithOnlyId = new ProgrammingSubmission();
-            }
-            else {
-                return;
+            switch (originalSubmission) {
+                case TextSubmission ignored -> submissionWithOnlyId = new TextSubmission();
+                case ModelingSubmission ignored -> submissionWithOnlyId = new ModelingSubmission();
+                case FileUploadSubmission ignored -> submissionWithOnlyId = new FileUploadSubmission();
+                case ProgrammingSubmission ignored -> submissionWithOnlyId = new ProgrammingSubmission();
+                default -> {
+                    return;
+                }
```

Comment:
Not sure of the extact synatx, but would something like this work?
```suggestion
            Submission submissionWithOnlyId = switch (originalSubmission) {
                case TextSubmission ignored -> new TextSubmission();
                case ModelingSubmission ignored -> submissionWithOnlyId = new ModelingSubmission();
                case FileUploadSubmission ignored -> new FileUploadSubmission();
                case ProgrammingSubmission ignored ->  new ProgrammingSubmission();
                default -> {
                    return;
                }
```

Reply:
This would be possible, but you forgot to remove `submissionWithOnlyId =` from the `ModelingSubmission` case and then the `default` case with the `return` doesn't work anymore, as the `switch` requires a valid return value from each path.

So, you could change it to this:

```suggestion
            Submission submissionWithOnlyId = switch (originalSubmission) {
                case TextSubmission ignored -> new TextSubmission();
                case ModelingSubmission ignored -> new ModelingSubmission();
                case FileUploadSubmission ignored -> new FileUploadSubmission();
                case ProgrammingSubmission ignored -> new ProgrammingSubmission();
                default -> null;
            };

            if (submissionWithOnlyId == null) {
                return;
            }
```

I actually think I like this even more, despite the extra if-check. What's your opinion?